### PR TITLE
4.x: Support for OKE Workload identity in OCI integration for Service registry

### PR DESCRIPTION
**Pull request by tomas-langer**
### Description
Adds OKE workload authentication details provider
Extracted some authentication details providers to separate modules (resource and instance principal)
Marked as preview feature (production ready, may have some changes) for service registry

A few fixes in Builder and Config metadata (+ test update)

### Documentation
Documentation of service registry features is now only in readme files

### CODE REVIEW

**Comment by arjav-desai** [comment]
I have a general nitpick that we should use full name i.e. authentication instead of atn, all over.

**Review by klustria — APPROVED** [review]
Other than comments from Arjav, in general, the changes LGTM. Mainly, I've looked at the OCI changes, and I'm good with those.

File: builder/codegen/src/main/java/io/helidon/builder/codegen/Types.java [hunk 1]
```diff
@@ -31,6 +32,8 @@ final class Types {
     static final TypeName ARRAY_LIST = TypeName.create(ArrayList.class);
     static final TypeName LINKED_HASH_SET = TypeName.create(LinkedHashSet.class);
     static final TypeName CHAR_ARRAY = TypeName.create(char[].class);
+    static final TypeName PATH = TypeName.create(Path.class);
+    static final TypeName URI = TypeName.create(java.net.URI.class);
```
[the remark this answers is not in the document]
Comment (tomas-langer) [reply builder/codegen/src/main/java/io/helidon/builder/codegen/Types.java:36]: Because it conflicts with the constant name

**Comment by tomas-langer** [comment]
I have resolved all comments.

**Review by Verdent — APPROVED** [review]

**Pull request merged**